File: WAStatusSaver/app/src/main/java/com/karts/wastatussaver/SplashActivity.java
package com.karts.wastatussaver;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;
import androidx.core.app.ActivityCompat;

import android.Manifest;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import com.karts.wastatussaver.util.SharedPrefs;
import com.karts.wastatussaver.util.Utils;

public class SplashActivity extends AppCompatActivity {

    String[] permissionsList = new String[]{Manifest.permission.READ_EXTERNAL_STORAGE,
            Manifest.permission.WRITE_EXTERNAL_STORAGE};

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        AppCompatDelegate.setDefaultNightMode(SharedPrefs.getAppNightDayMode(this));

        Utils.setLanguage(SplashActivity.this, SharedPrefs.getLang(SplashActivity.this));


        if (Utils.hasPermissions(this, permissionsList)) {
            ActivityCompat.requestPermissions(this,  Utils.permissions, Utils.perRequest);
        }else {
            gotoNext();
        }


    }

    void gotoNext(){
        new Handler().postDelayed(() -> {
            startActivity(new Intent(SplashActivity.this, MainActivity.class).addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP));
            finish();
        }, 600);

    }

    @Override
    public void onBackPressed() { }


    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        if (requestCode == Utils.perRequest) {
            if (Utils.hasPermissions(this, permissionsList)) {
                ActivityCompat.requestPermissions(this,  Utils.permissions, Utils.perRequest);
            }else {
                gotoNext();
            }
        }
    }


}

File: XamppLocal/XamppLocal/app/src/main/java/com/example/xampplocal/SplashActivity.kt
package com.example.xampplocal

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import com.example.xampplocal.sharepref.PrefManager

class SplashActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        Handler().postDelayed({
            var prefManager = PrefManager(this)
            if(!prefManager.getLoginStatus()){
                startActivity(Intent(this,LoginActivity::class.java))
                finish()

            }else{
                if(prefManager.getLoginStatus()){
                    startActivity(Intent(this,HomeActivity::class.java))
                    finish()

                }/*else{
                    startActivity(Intent(this,MainActivity::class.java))
                }*/
            }
        },3000)
    }
}